### ISSUE TITLE
Nuget dependency chaining causing unwanted dependency on Fody

### ISSUE THREAD

**Issue by twaddell** [csnemes/tracer#94]
I am building a dll project into a Nuget package. This dll project references the Tracer.4NLog.Fody package for it's logging.

If I include my Nuget package in a project an unwanted reference to Fody is included and I'm having to include FodyWeaver.xml etc. in every project that uses my package.

This seems unnecessary to me as Fody is only needed at compile time in the original dll project and I don't believe it should be required in projects referencing my Nuget package.

I have tried setting the compile time references to be PrivateAssets as follows:
```
<PackageReference Include="Tracer.4NLog.Fody" Version="3.2.0">
  <PrivateAssets>build; native; contentfiles; analyzers</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
```
but there is still a dependency on Fody.

I think essentially I am looking for a way to reference the Tracer.NLog.dll without also referencing Fody.

**Comment by csnemes** [comment]
In order to use Tracer.4NLog.Fody properly the assembly referencing it should be rewritten. So I cant see exactly how you can use the Tracer library without using Fody.

**Comment by twaddell** [comment]
I agree that a Fody reference is needed for the project immediately referencing Tracer.4NLog.Fody but the dependency on Fody continues up the project chain.
For example, we are creating `MyLibrary` and want to include tracing functionality for the library so add the Nuget package for `Tracer.4NLog.Fody`. `MyLibrary` now includes `Fody` because of the following dependency chain.
`MyLibrary` -> `Tracer.4NLog.Fody` -> `Fody`

However the reference to `Fody` is not set as PrivateAssets so when `MyApp` references `MyLibrary` it now also has an unnecessary dependency on `Fody`
`MyApp` -> `MyLibrary` -> `Tracer.4NLog.Fody` -> `Fody`

Is there a way to set the dependency on `Fody` to not keep passing up the dependecy chain? Or do I need to set an explicit reference to `Fody` in `MyLibrary` with PrivateAssets set?
